packages/components/src/slot-fill/slot.tsx

```diff
@@ -32,41 +32,48 @@ function isFunction( maybeFunc: any ): maybeFunc is Function {
 	return typeof maybeFunc === 'function';
 }
 
+function addKeysToChildren( children: ReactNode ) {
+	return Children.map( children, ( child, childIndex ) => {
+		if ( ! child || typeof child === 'string' ) {
+			return child;
+		}
+		let childKey: Key = childIndex;
+		if ( typeof child === 'object' && 'key' in child && child?.key ) {
+			childKey = child.key;
+		}
+
+		return cloneElement( child as ReactElement, {
+			key: childKey,
+		} );
+	} );
+}
```

Comment:
This is a simple refactor where the `addKeysToChildren` function has been extracted from the big `Slot` component. Without any change.